extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -1272,8 +1272,7 @@ public Locale getLanguageInfo() {
                 final String panelIdentifier = panels.getObject(i)
                         .getObject("engagementPanelSectionListRenderer")
                         .getString("panelIdentifier");
-                if (panelIdentifier.equals(
-                        "engagement-panel-macro-markers-description-chapters")) {
+                if ("engagement-panel-macro-markers-description-chapters".equals(panelIdentifier)) {
```

Comment:
It's not obvious at all why the comparison is flipped if you don't know `panelIdentifier` can be `null`. How about adding a comment?

Reply:
Something like that?
``` java
                // panelIdentifier might be null if the panel has something to do with ads
                // See https://github.com/TeamNewPipe/NewPipe/issues/7792#issuecomment-1030900188
                @Nullable
                if ("engagement-panel-macro-markers-description-chapters".equals(panelIdentifier)) {
```

Reply:
because that var can be null

Reply:
Ah, I saw the comment before you edited it; the `@Nullable` was directly before the `if` statement then. xD